README.md

```diff
@@ -1,4 +1,74 @@
 # OPA Spring Security Library
 
+OPA Spring Security is a library that enables usage of OPA as authorization tool in Spring applications.
 
+## Installation
 
+#### Prerequisites 
+
+- Java 11 or higher.
+- Your project has to have `org.springframework.security:spring-security-oauth2-jose` on classpath  
+
+#### Using the starter
+
+Add dependency using Maven
+
+```xml
+<dependency>
+    <groupId>com.bisnode.opa</groupId>
+    <artifactId>opa-filter-spring-boot-starter</artifactId>
+    <version>{version}</version>
+</dependency>
+```
+
+or Gradle
+
+```groovy
+implementation 'com.bisnode.opa:opa-filter-spring-boot-starter:{version}'
+```
+
+## Configuration
+
+**All properties are prefixed with `opa.filter`**
+
+| Property   | Default value | Description                           | Example                    |
+|------------|---------------|---------------------------------------|----------------------------|
+| `enabled`  |    `true`     | Whether the filter should be enabled  | _false_                    |
+| `policy`   |               | Name of OPA policy to use for queries | _"http/request/authz"_     |
```

Comment:
Is this the path to the rule? I think we should consider naming here if so.

Reply:
It's the rule's `package`, I don't really know how to name it. Any suggestions?

Reply:
But to the client it's only used to build the request URL, so `instance` + `policy` builds the complete URL, right? It's technically querying the `data` API, so this might get confusing. But we could review naming separately before first real release.

Reply:
Ok, I'm even more confused now. @irodzik ?

Reply:
If we want to go for plain OPA language here, we are querying for document https://www.openpolicyagent.org/docs/latest/rest-api/#query-api.
As OPA query basically is a tool for computing/evaluating based on some rules and data. In our case we use it to compute authorization decision. From what i understand, document under some path is a mix of data and rules. When you query for document with your input, data is merged with input and evaluated using rules. @anderseknert correct me if i am wrong.
In OPA terminology, these documents are lying under some path (like on server).

Reply:
so... `documentPath`?

Reply:
Sounds good to me!